match.go

```diff
@@ -67,14 +68,47 @@ type globMatcher struct {
 }
 
 func (m *globMatcher) Match(name string) bool {
-	matches, err := filepath.Match(m.glob, name)
+	matches, err := doublestar.PathMatch(m.glob, name)
 	if err != nil {
 		return false
 	}
 	return matches != m.inverse
 }
 
-func (m *globMatcher) ExcludePrefix(prefix string) bool { return false }
+func (m *globMatcher) ExcludePrefix(prefix string) bool {
+	if !m.inverse || len(m.glob) == 0 {
+		return false
+	}
+
+	matches, err := doublestar.PathMatch(m.glob, prefix)
+	switch {
+	case err != nil:
+		return false
+	case matches:
+		return true
```

Comment:
The documentation for the ExcludePrefix method says:

```
// ExcludePrefix returns whether all paths with this prefix cannot match.
// It is allowed to return false negatives but not false positives.
```

That is not what this function implements. For example, if the pattern is `**/`, that matches the prefix `a/` but does not match the path `a/b.txt`. So this would incorrectly exclude `a/b.txt` from the watch.

It's not immediately obvious to me how to fix this problem. I designed the ExcludePrefix mechanism mostly with regexp matching in mind.